be/src/vec/exec/format/table/iceberg_reader.cpp

```diff
@@ -134,17 +166,32 @@ Status IcebergTableReader::init_row_filters() {
                 ParquetReader* delete_reader = new ParquetReader(
                         _profile, _params, delete_range, _state->query_options().batch_size,
                         const_cast<cctz::time_zone*>(&_state->timezone_obj()));
-                FileMetaData* metadata = nullptr;
-                RETURN_IF_ERROR(delete_reader->file_metadata(&metadata));
-
-                auto& delete_file_schema = metadata->schema();
+                if (_delete_file_schema == nullptr) {
+                    FileMetaData* metadata = nullptr;
+                    RETURN_IF_ERROR(delete_reader->file_metadata(&metadata));
+                    if (metadata == nullptr) {
+                        break;
+                    }
+                    _delete_file_schema.reset(&metadata->schema());
+                    int num_of_col = _delete_file_schema->size();
+                    for (auto i = 0; i < num_of_col; ++i) {
+                        const FieldSchema* field = _delete_file_schema->get_column(i);
+                        _column_schemas.emplace_back(field);
+                    }
+                }
+                DCHECK_EQ(_column_schemas.size(), _delete_file_schema->size());
                 vector<std::string> names;
-                for (auto i = 0; i < delete_file_schema.size(); ++i) {
-                    const FieldSchema* field = delete_file_schema.get_column(i);
-                    _column_schemas.emplace_back(field);
+                for (const FieldSchema* field : _column_schemas) {
```

Comment:
This `for (const FieldSchema* field : _column_schemas)` can also be done only once.
Because all delete files have same column names.